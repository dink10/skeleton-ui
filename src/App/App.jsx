import React from 'react';
import PropTypes from 'prop-types';
import Button from 'gismart-ui/core/components/Button';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

function Home() {
  return (
    <div>
      <ul>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/about">About</Link>
        </li>
        <li>
          <Link to="/topics">Topics</Link>
        </li>
        <Button>A</Button>
      </ul>
    </div>
  );
}

function About() {
  return (
    <div>
      <h2>About</h2>
    </div>
  );
}

function Topic({ match }) {
  return (
    <div>
      <h3>{match.params.topicId}</h3>
    </div>
  );
}

Topic.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      topicId: PropTypes.string.isRequired,
    }),
  }).isRequired,
};

function Topics({ match }) {
  return (
    <div>
      <h2>Topics</h2>
      <ul>
        <li>
          <Link to={`${match.url}/rendering`}>Rendering with React</Link>
        </li>
        <li>
          <Link to={`${match.url}/components`}>Components</Link>
        </li>
        <li>
          <Link to={`${match.url}/props-v-state`}>Props v. State</Link>
        </li>
      </ul>

      <Route path={`${match.path}/:topicId`} component={Topic} />
      <Route
        exact
        path={match.path}
        render={() => <h3>Please select a topic.</h3>}
      />
    </div>
  );
}

Topics.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string.isRequired,
    path: PropTypes.string.isRequired,
  }).isRequired,
};

export default function App() {
  return (
    <Router>
      <Route exact path="/" component={Home} />
      <Route path="/about" component={About} />
      <Route path="/topics" component={Topics} />
    </Router>
  );
}
