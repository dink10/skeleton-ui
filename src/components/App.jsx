import React from 'react'
// import PropTypes from 'prop-types';
import { ConnectedRouter } from 'connected-react-router'
import { Switch, Route, Redirect } from 'react-router-dom'
import history from '/history'
import Layout, { Footer, Content } from 'gismart-ui/core/components/Layout'
import Header from './Header'

import { Login, NotFound } from '/pages'
import Secure from './Secure'

function App() {
  return (
    <ConnectedRouter history={history}>
      <Layout>
        <Header />
        <Content>
          <Route exact path="/" render={() => <Redirect to="/home" />} />
          <Switch>
            <Route path="/login" component={Login} />
            <Route
              path="/home"
              render={() => (
                <Secure>
                  <div>Hello home</div>
                </Secure>
              )}
            />
            <Route component={NotFound} />
          </Switch>
        </Content>
        <Footer>Â©2019 Created by GISMART</Footer>
      </Layout>
    </ConnectedRouter>
  )
}

export default App
