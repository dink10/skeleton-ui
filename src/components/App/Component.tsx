import React from 'react'
// import { ConnectedRouter } from 'connected-react-router'
// import { Switch, Route, Redirect } from 'react-router-dom'
// import history from '/history'
// import { BackToTop } from 'gismart-ui/core/components'
// import Layout, {
//   Footer, Content, MainWrapper, SubHeader,
// } from 'gismart-ui/core/components/Layout'
// import Header from './Header'

// import { Login, NotFound } from '/pages'
// import Secure from './Auth'
// import Spinner from './Spinner'
// import ErrorNotification from './ErrorNotification'
// import Breadcrumb from './Breadcrumb'

function App() {
  return (
    <div>Hello type script test hot reload 123 432 0999 999 333</div>
  )
}

export default App
