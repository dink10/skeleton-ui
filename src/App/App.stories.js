/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import App from './App';

export default { title: 'App' };

export const DefaultApp = () => <App />;
