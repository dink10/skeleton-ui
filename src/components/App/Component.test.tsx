import * as React from 'react'
import { create, ReactTestRenderer } from 'react-test-renderer'
import App from './Component'

test('app snapshot', () => {
    const component: ReactTestRenderer = create(<App />)
    expect(component).toMatchSnapshot()
    component.unmount()
})
