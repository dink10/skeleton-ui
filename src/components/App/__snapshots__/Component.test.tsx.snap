// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshot 1`] = `
<div>
  Hello type script test hot reload 123 432 0999 999 333
</div>
`;
